### PR TITLE
update the syntax for affinity labels; also, fix regression

### PR DESCRIPTION
**Pull request by sonchang**
There was a regression causing a TF in test_service_affinity_rules.  Basically, user's custom service labels wasn't being populated in instances.
@ibuildthecloud @alena1108

### CODE REVIEW

File: code/iaas/allocator/src/main/java/io/cattle/platform/allocator/constraint/ContainerLabelAffinityConstraint.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 public class ContainerLabelAffinityConstraint implements Constraint {
     public static final String ENV_HEADER_AFFINITY_CONTAINER_LABEL = "affinity:container_label:";
-    public static final String LABEL_HEADER_AFFINITY_CONTAINER_LABEL = "io.rancher.scheduler." + ENV_HEADER_AFFINITY_CONTAINER_LABEL;
+    public static final String LABEL_HEADER_AFFINITY_CONTAINER_LABEL = "io.rancher.scheduler.affinity:container_label";
```
Comment (ibuildthecloud) [inline code/iaas/allocator/src/main/java/io/cattle/platform/allocator/constraint/ContainerLabelAffinityConstraint.java]: @sonchang @vincent99 Just need a consensus.  Are we going with `container-label` or `container_label`. (same applies for host_label).  I void for `container-label`

**Comment by ibuildthecloud** [comment]
I'm going to defer to @alena1108 to review the deployment unit changes.

File: code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentUnit.java [hunk 2]
```diff
@@ -194,6 +194,10 @@ public boolean isComplete() {
     protected Map<String, String> getLabels(DeploymentUnitInstance instance) {
         Service service = instance.getService();
         Map<String, String> labels = new HashMap<>();
+
+        // TODO: This might change in the future if we merge the labels from all services
+        labels.putAll(context.sdService.getServiceLabels(service));
```
Comment (alena1108) [inline code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentUnit.java]: No need to populate the labels here. Service labels get set on the instance in ServiceDiscoveryServiceImpl.buildServiceInstanceLaunchData call, along with other parameters defined on the service level. In this code only deployment unit related labels get populated.

Comment (alena1108) [inline code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentUnit.java]: But I see why you need these labels here as you operate on them later in the code. So should be fine to leave it here for now.

Comment (sonchang) [inline code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentUnit.java]: Ok.  Just took a look.  The changes are safe, but not ideal.  I'll update for better consistency.  Thanks!

**Comment by alena1108** [comment]
LGTM

**Comment by ibuildthecloud** [comment]
LGTM  I'll merge when build finishes

**Pull request merged**